docs/devs/plugin-migration-to-alpha-14.md

```diff
@@ -0,0 +1,523 @@
+```
+
+**Alpha 14 (JS)**
+```js
+const { BaseControllerPlugin } = require("@clusterio/controller");
+const { BaseInstancePlugin } = require("@clusterio/host");
+```
+
+**Alpha 14 (TS)**
+```ts
+import { BaseControllerPlugin } from "@clusterio/controller";
+import { BaseInstancePlugin } from "@clusterio/host";
+```
+
+
+## Change main entrypoint of plugin
+
+Previously the info of a plugin would be loaded by requiring `./plugin_dir/info`, this has been changed to require `./plugin_dir` which in turn means that `info.js` is no longer the default entrypoint.
+To solve this you can either add `"main": "info.js",` to package.json or rename `info.js` to `index.js` (the rest of this guide assumes the entrypoint was renamed to `index.js`).
+The motivation behind this change was to allow TypeScript to output its build to a sub-folder of the plugin, if you're using TypeScript you'll likely have the entrypoint be located in `dist/index.js` instead.
+
+The expected place the declaration of the plugin is exported also changed from `module.exports` to `module.exports.plugin`:
+
+**Alpha 13**
+```js
+// info.js
+module.exports = {
+    name: "plugin_name",
+    ...
+}
+```
+
+**Alpha 14 (JS)**
+```js
+// index.js
+const plugin = {
+    name: "plugin_name",
+    ...
+}
+
+module.exports = {
+    plugin,
+};
+```
+
+**Alpha 14 (TS)**
+```ts
+// index.ts
+import type * as lib from "@clusterio/lib";
+
+export const plugin: lib.PluginDeclaration = {
+    name: "plugin_name",
+    ...
+};
+```
+
+
+### Modify exposed path in webpack.config.js
+
+The `./info` import path exposed in the ModuleFederationPlugin in the `webpack.config.js` needs to be changed to `./`.
+E.g.:
+
+**Alpha 13**
+```js
+// In webpack.config.js
+new webpack.container.ModuleFederationPlugin({
+    ...
+    exposes: {
+        "./info": "./info.js",
+        ...
+    },
+}),
+```
+
+**Alpha 14**
+```js
+// In webpack.config.js
+new webpack.container.ModuleFederationPlugin({
+    ...
+    exposes: {
+        "./": "./index.js",
+        ...
+    },
+}),
+```
+
+## Remove usage of ConfigGroup
+
+The Config system was refactored and the `ConfigGroup` concept was completely removed.
+Config fields are now defined directly on the exported plugin info.
+
+**Alpha 13**
+```js
+const lib = require("@clusterio/lib");
+
+class MasterConfigGroup extends lib.PluginConfigGroup { }
+MasterConfigGroup.groupName = "plugin_name";
+MasterConfigGroup.defaultAccess = ["master", "control"];
+MasterConfigGroup.define({
+    name: "option",
+    title: "Option",
+    description: "A config option.",
+    type: "string",
+    optional: true,
+});
+
+module.exports = {
+    name: "plugin_name",
+    MasterConfigGroup,
+    ...
+}
+```
+
+**Alpha 14 (JS)**
+```js
+const plugin = {
+    name: "plugin_name",
+    controllerConfigFields: {
+        "plugin_name.option": {
+            title: "Option",
+            description: "A config option.",
+            type: "string",
+            initial_value: "string",
```

Comment:
Using `"string"` here might have people assume it is the type of the inital value. I recomend something like "foo" or "initial option" (same applies to ts version)